### ISSUE TITLE
BigBiGAN Loss Function

### ISSUE THREAD

**Issue by shreejalt** [tensorflow/tensorflow#50674]
Hi, 
I wanted to ask whether the BigBiGAN network contains KL Divergence Loss in the implementation? Because they have mentioned non-deterministic encoder architecture for sampling z vector. They used Variational Bayes for sampling as mentioned.

If anyone can help me out, it would be great!

Thanks in advance.

**Comment by Saduf2019** [comment]
@shreejalt 
Can you please let us know, where is it mentioned about "non-deterministic encoder architecture for sampling z vector".

**Comment by shreejalt** [comment]
@Saduf2019 , 

As in ALI [10], the encoder E of our Base model is nondeterministic, parametrizing a distribution N (µ, σ). µ and σˆ are given by a linear layer at the output
of the model, and the final standard deviation σ is computed from σˆ using a non-negative “softplus”
non-linearity σ = log(1 + exp(ˆσ)) [9]. The final z uses the reparametrized sampling from [23],
with z = µ + σ, where  ∼ N (0, I). Compared to a deterministic encoder (row Deterministic E)
which predicts z directly without sampling (effectively modeling P(z|x) as a Dirac δ distribution),
the non-deterministic Base model achieves significantly better classification performance. 

This is copy-pasted from the paper.  Ablation 3.1 (latent distribution Pz and stochastic E). 


Let me know if i am getting something wrong...

Paper Link: https://arxiv.org/pdf/1907.02544.pdf

**Comment by Saduf2019** [comment]
@shreejalt 
Where in tensorflow do you expect these changes to be made.
